### ISSUE TITLE
Version 4.4 cannot determine partition type

### ISSUE THREAD

**Issue by markmckinnon** [sleuthkit/sleuthkit#769]
When trying to open the sample image http://digitalcorpora.org/corp/nps/scenarios/2009-m57-patents/drives-redacted/charlie-2009-12-11.E01 with mmls (version 4.4.0) I get the following error "Cannot determine partition type (GPT or DOS at 0)".  Opening the same image with mmls (Version 4.3.0) I am able to open and display the contents of the volume system.  If there is more information I can provide please let me know.

Kind regards.

Mark

**Comment by bcarrier** [comment]
I think the only change to the VS code was to look for a backup GPT partition table so that we can support some Android images that have a bootloader in the beginning.  (https://github.com/sleuthkit/sleuthkit/commit/ff0f02008bdf78a4d9e9ebda7dbd942b9f993670). 

Ann is going to take a look at this image to see if that is the case.  Thanks.

**Comment by vicodark** [comment]
I have the exact same issue on a drive image I can't share, but could dd the first few sectors or whatever is needed for testing.

**Comment by bcarrier** [comment]
We confirmed that the NPS image has a DOS partition table in the beginning of the drive and a backup GPT partition table at the end and it wants the user to decide which is correct.  If you specify 'mmls -t dos', then it will use the DOS partition table. 

I think we need to modify the code to let the DOS partition table trump the backup GPT and not consider it a conflict.

**Comment by vicodark** [comment]
Ah, good. Even just a more informative error message would be nice, as in "Cannot determine partition type (GPT or DOS at 0), please specify which with -t ... "

**Comment by jtsylve** [comment]
An error message won't help those who just use the library.  I agree with @bcarrier, the auto detect stuff needs to only attempt backup GPT after all other options have failed

**Comment by kefir-** [comment]
I second that. For cases like this, TSK should mirror whatever common operating systems do by default, and let the user override as required.

For what it's worth, Linux simply uses the dos partition table for this image:
```
$ fdisk -l /dev/sdb
Disk /dev/sdb: 9,5 GiB, 10239860736 bytes, 19999728 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
Disklabel type: dos
Disk identifier: 0xeaa9eaa9

Device     Boot Start      End  Sectors  Size Id Type
/dev/sdb1  *       63 19968794 19968732  9,5G  7 HPFS/NTFS/exFAT

Partition 1 does not start on physical sector boundary.
```

**Comment by bcarrier** [comment]
This was fixed a while back in #788.
